### PR TITLE
[IMP] *: wrong any() usage

### PR DESCRIPTION
**Pull request by Feyensv**
Replace wrong usages of any(list|recordset), by any(generator)
to speed up computations, avoiding list creations and/or looping twice on a recordset
for nothing.

any([generator]) => any(generator)
any(filtered) => any(generator)

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
Seems like a good idea. Added in my todo yay.

**Comment by Feyensv** [comment]
Well, I planned a quick merge but if you want to supervise it, it can wait a bit :thinking:.

**Comment by tde-banana-odoo** [comment]
Ow, ok for it if you go for a quick merge.

File: addons/mrp_subcontracting/wizard/stock_picking_return.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def _compute_subcontract_location_id(self):
     @api.onchange('picking_id')
     def _onchange_picking_id(self):
         res = super(ReturnPicking, self)._onchange_picking_id()
-        if any(self.product_return_moves.filtered(lambda r: r.quantity > 0).move_id.mapped('is_subcontract')):
+        if any(wizard_line.quantity > 0 and wizard_line.move_id.is_subcontract for wizard_line in self.product_return_moves):
```
Comment (mart-e) [inline addons/mrp_subcontracting/wizard/stock_picking_return.py:22]: detail, `wizard_line` is not very clear (technical), prefer something explaining the meaning of the record, e.g. `return_line`

File: addons/stock/report/report_stockpicking_operations.xml [hunk 2]
```diff
@@ -59,7 +59,7 @@
                                 </div>
                             </div>
                             <table class="table table-sm" t-if="o.move_line_ids and o.move_ids_without_package">
-                                <t t-set="has_barcode" t-value="any([move_ids_without_package.product_id and move_ids_without_package.product_id.sudo().barcode or move_ids_without_package.package_id for move_ids_without_package in o.move_line_ids])"/>
+                                <t t-set="has_barcode" t-value="any(ml.product_id and ml.product_id.sudo().barcode or ml.package_id for ml in o.move_line_ids)"/>
```
Comment (mart-e) [inline addons/stock/report/report_stockpicking_operations.xml:62]: don't fall for the `product.ptav`, bytes are cheap, it's ok to use them

**Comment by Feyensv** [comment]
@robodoo r+

**Comment by Feyensv** [comment]
@robodoo r+

**Pull request closed without merging**